### PR TITLE
misc: join all S3 Express async refresh jobs in tests

### PR DESCRIPTION
**Pull request by lauzadis**
Fixes flaky tests by joining all async refresh jobs before closing the DefaultS3ExpressCredentialsProvider

## Description of changes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: services/s3/common/src/aws/sdk/kotlin/services/s3/express/DefaultS3ExpressCredentialsProvider.kt [hunk 1]
```diff
@@ -71,6 +73,10 @@ internal class DefaultS3ExpressCredentialsProvider(
         coroutineContext.cancel(null)
     }
 
+    internal suspend fun waitForShutdown() {
```
Comment (aajtodd) [inline services/s3/common/src/aws/sdk/kotlin/services/s3/express/DefaultS3ExpressCredentialsProvider.kt:76]: Shouldn't be necessary. You can just join the scope.

```kotlin
provider.coroutineContext.job.join()
```

**Review by ianbotsf — APPROVED** [review]

File: services/s3/common/src/aws/sdk/kotlin/services/s3/express/DefaultS3ExpressCredentialsProvider.kt [hunk 2]
```diff
@@ -68,6 +68,7 @@ internal class DefaultS3ExpressCredentialsProvider(
     }
 
     override fun close() {
+        runBlocking { coroutineContext.job.join() }
```
Comment (aajtodd) [inline services/s3/common/src/aws/sdk/kotlin/services/s3/express/DefaultS3ExpressCredentialsProvider.kt:71]: **correctness**: `runBlocking` doesn't exist in JS FYI. This appears to only be used in tests? Is there a reason to not just have the tests call `join()` on their own?

**Review by aajtodd — APPROVED** [review]
Approved with alternative suggestion.

File: services/s3/common/src/aws/sdk/kotlin/services/s3/express/DefaultS3ExpressCredentialsProvider.kt (hunk 2 above)
Comment (lauzadis) [reply services/s3/common/src/aws/sdk/kotlin/services/s3/express/DefaultS3ExpressCredentialsProvider.kt:71]: The behavior is currently only used in tests, but I think having all async refreshes be completed before the provider is marked as closed is the safest behavior.

**Pull request merged**